src/Currencies/CurrencyList.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Money\Currencies;
+
+use Money\Currencies;
+use Money\Currency;
+use Money\Exception\UnknownCurrencyException;
+
+/**
+ * A list of custom currencies.
+ *
+ * @author George Mponos <gmponos@gmail.com>
+ */
+final class CurrencyList implements Currencies
+{
+    /**
+     * Map of currencies indexed by code.
+     *
+     * @var array
+     */
+    private $currencies;
+
+    public function __construct(array $currencies)
+    {
+        foreach ($currencies as $currencyCode => $currency) {
+            if (empty($currencyCode) || !is_string($currencyCode)) {
+                throw new \InvalidArgumentException(
+                    sprintf('Currency code must be a string and not empty. "%s" given', $currencyCode)
+                );
+            }
+
+            if (
+                !isset($currency['minorUnit']) ||
```

Comment:
See my eariler comment regarding this.

Reply:
I believe it's fine now. Check it out :)